### PR TITLE
Leave 'post_fail_hook' early if 'NOLOGS' is set

### PR DESCRIPTION
**Pull request by rfan1**
https://progress.opensuse.org/issues/120729

For some reason, somebody might not want to collect logs, but still run the post_fail_hooks, and we only have that for opensusebasetest but not for subclasses of it
This will help to run the post_fail_hooks of specific test modules that are also calling $self->SUPER::post_fail_hook where collecting those extra logs isn't really necessary

- Related ticket: https://progress.opensuse.org/issues/120729
- Verification run: n/a

### CODE REVIEW

**Comment by Zaoliang** [comment]
LGTM

**Review by pdostal — DISMISSED** [review]
This is big change. I would definitely inform wider group of people about this change.

**Comment by rfan1** [comment]
@foursixnine 
Could you please involve more experts to review this PR?

File: lib/apparmortest.pm [hunk 1]
```diff
@@ -802,6 +802,8 @@ Run post_fail_hook and upload audit logs
 sub post_fail_hook {
     my ($self) = shift;
 
+    return if (get_var('NOLOGS'));
+
```
Comment (okurz) [inline lib/apparmortest.pm:806]: Please use just `return if get_var('NOLOGS')` without the additional pair of parentheses as there is only a single argument.

File: lib/bootbasetest.pm [hunk 2]
```diff
@@ -7,6 +7,7 @@ use warnings;
 
 sub post_fail_hook {
     my ($self) = @_;
+
```
Comment (okurz) [inline lib/bootbasetest.pm:10]: unrelated change

File: lib/y2_base.pm [hunk 3]
```diff
@@ -160,6 +160,9 @@ sub upload_widgets_json {
 
 sub post_fail_hook {
     my $self = shift;
+
+    return if (get_var('NOLOGS'));
+
```
Comment (okurz) [inline lib/y2_base.pm:165]: I consider this too verbose. Just leave out the blank lines please. This applies for all changes

**Review by okurz — CHANGES_REQUESTED** [review]

File: lib/apparmortest.pm (hunk 1 above)
Comment (rfan1) [reply lib/apparmortest.pm]: Thanks @okurz 
fixed as your kindly suggestion.

**Review by foursixnine — APPROVED** [review]
Looks good for me, @os-autoinst/tests-maintainer any comments or objections?

**Review by alvarocarvajald — APPROVED** [review]
LGTM

**Review by jlausuch — APPROVED** [review]

**Review by dzedro — APPROVED** [review]

**Review by ge0r — APPROVED** [review]
LGTM

**Comment by lemon-suse** [comment]
It's a good idea and save time for debug. If we can find easy method to apply this to overwritten post_fail_hook, then that will be perfect. :)

Just please update the PR description and remove things from template if you don't need them any more.

**Review by okurz — APPROVED** [review]
